woltka/workflow.py

```diff
@@ -0,0 +1,590 @@
+    else:
+        return {files[fp]: rmap}
+
+
+def assign_rmap(rmap:     dict,
+                data:     dict,
+                rank:     str,
+                sample:   str,
+                rank2dir: dict = None,
+                tree:     dict = None,
+                rankd:    dict = None,
+                named:    dict = None,
+                root:      str = None,
+                above:    bool = False,
+                major:     int = None,
+                ambig:     str = True,
+                subok:    bool = None):
+    """Assign query sequences in a query-to-subjects map to classification
+    units based on their subjects.
+
+    Parameters
+    ----------
+    rmap : dict
+        Read map to assign.
+    data : dict
+        Master data structure.
+    rank : str
+        Target rank to assign to.
+    sample : str
+        Sample ID.
+    rank2dir : dict, optional
+        Directory of output maps per rank.
+
+    tree : dict, optional
+        Hierarchical classification system.
+    rankd : dict, optional
+        Rank dictionary.
+    named : dict
+        Taxon name directory.
+    root : str, optional
+        Root identifier.
+
+    above : bool, optional
+        Assignment above given rank is acceptable (for fixed ranks).
+    major : float, optional
+        Majority-rule assignment threshold (available only with a fixed rank
+        and not above or ambig).
+    ambig : bool, optional
+        Count occurrence of each possible assignment instead of targeting one
+        assignment (available only with a fixed rank and not above).
+    subok : bool, optional
+        Allow assignment to subject(s) itself instead of higher classification
+        units.
+    """
+    # determine assigner
+    if rank is None or rank == 'none' or tree is None:
+        assigner = assign_none
+        args = (ambig,)
+    elif rank == 'free':
+        assigner = assign_free
+        args = (tree, root, subok)
+    else:
+        assigner = assign_rank
+        args = (rank, tree, rankd, root, above, major, ambig)
+
+    # call assigner
+    asgmt = {k: assigner(v, *args) for k, v in rmap.items()}
+
+    # write classification map
+    if rank2dir is not None:
+        with open(join(rank2dir[rank], f'{sample}.tsv'), 'a') as f:
+            write_map(f, asgmt, named)
+
+    # count taxa
+    counts = count(asgmt)
+
+    # round floats
+    intize(counts)
+
+    # delete "None" keys
+    delnone(counts)
+
+    # combine old and new counts
+    if sample in data[rank]:
+        add_dict(data[rank][sample], counts)
+    else:
+        data[rank][sample] = counts
+
+
+def write_profiles(path_:    str,
+                   data:    dict,
+                   named:   dict = None,
+                   samples: list = None):
+    """Write profile to a tab-delimited file.
+
+    Parameters
+    ----------
+    data : dict
+        Profile data.
+    path_ : str
+        Path to output file or directory.
+    named : dict, optional
```

Comment:
```suggestion
    name_dict : dict, optional
```

Reply:
"named" is a valid english word with a significantly different meaning than "name dictionary"

Reply:
Agreed and changed! (But somehow in the middle, as `namedic` :))